masonry_winit/examples/calc_masonry.rs

```diff
@@ -283,10 +284,7 @@ impl AppDriver for CalcState {
         }
 
         ctx.render_root().edit_root_widget(|mut root| {
-            let mut root = root.downcast::<RootWidget>();
-            let mut flex = RootWidget::child_mut(&mut root);
-            let mut flex = flex.downcast();
-            let mut label = Flex::child_mut(&mut flex, 1).unwrap();
+            let mut label = root.ctx.find_mut(self.ids.display);
             let mut label = label.downcast::<Label>();
             Label::set_text(&mut label, &*self.value);
         });
```

Comment:
Note that if you already know the id of the widget you're trying to mutate, you don't need a new method:

```rust
         ctx.render_root().edit_widget(self.ids.display, |mut label| {
            let mut label = label.downcast::<Label>();
            Label::set_text(&mut label, &*self.value);
         };
```

Reply:
By the way, my medium-term plan for this kind of use-case is to create a new kind of string id (something like `WidgetKey`) that could be assigned to a widget at construction and let you identify them, basically an equivalent of the DOM `id` attribute.

Reply:
> you don't need a new method

Oh right yeah. I'll create a new PR to use that in the examples.

> my medium-term plan for this kind of use-case is to create a new kind of string id

I really don't see how that would be better than just using WidgetIds. If we have sth else I don't think it should be strings because of type-safety.